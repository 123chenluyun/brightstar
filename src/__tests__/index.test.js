import React from 'react';
import ReactDOM from 'react-dom';
import App from '../index';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';
import renderer from 'react-test-renderer';

describe('<App />', () => {

	let app, props;

	beforeEach(() => {
	  props = [
	  	{id:1, contributors_url:"www.test.com",  stargazers_count: 100, name: "test", description: "test description" },
	  	{id:2, contributors_url:"www.test2.com",  stargazers_count: 99, name: "test2", description: "test2 description" }
	  ];
	  app = shallow(<App />); 
	}); 

	it('renders without crashing', () => {
	  const div = document.createElement('div');
	  ReactDOM.render(<App {...props} />, div);
	});

	it('renders loader to start', () => {
	  const app = renderer.create(
	   <App />
	  ).toJSON();
	  expect(app).toMatchSnapshot();
	});

	it('renders repos and header after setting state', () => {
	  const app = shallow(<App />);
	  app.setState({
		  repos: [
		  	{id:1, contributors_url:"www.test.com",  stargazers_count: 100, name: "test", description: "test description" },
		  	{id:2, contributors_url:"www.test2.com",  stargazers_count: 99, name: "test2", description: "test2 description" }
		  ],
		  loaded: true
	  });
	  expect(app).toMatchSnapshot();
	});

	it('calls componentWillMount', () => {
	   sinon.spy(App.prototype, 'componentWillMount');
	   app = mount(<App {...props}/>); 
	   expect(App.prototype.componentWillMount.calledOnce).toEqual(true);
	 });

	it('Index has proper elements', () => {
		expect(app.find('.invisible').length).toEqual(1);
		expect(app.text()).toEqual("<Loader />");
	});

});