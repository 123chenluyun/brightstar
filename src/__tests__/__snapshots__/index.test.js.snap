// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders loader to start 1`] = `
<div>
  
  <div
    className="loader-container"
  >
    <div
      className="sk-circle"
    >
      <div
        className="sk-circle1 sk-child"
      />
      <div
        className="sk-circle2 sk-child"
      />
      <div
        className="sk-circle3 sk-child"
      />
      <div
        className="sk-circle4 sk-child"
      />
      <div
        className="sk-circle5 sk-child"
      />
      <div
        className="sk-circle6 sk-child"
      />
      <div
        className="sk-circle7 sk-child"
      />
      <div
        className="sk-circle8 sk-child"
      />
      <div
        className="sk-circle9 sk-child"
      />
      <div
        className="sk-circle10 sk-child"
      />
      <div
        className="sk-circle11 sk-child"
      />
      <div
        className="sk-circle12 sk-child"
      />
    </div>
  </div>
</div>
`;

exports[`<App /> renders repos and header after setting state 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Header />
    <RepoList
        repos={
            Array [
                Object {
                  "contributors_url": "www.test.com",
                  "description": "test description",
                  "id": 1,
                  "name": "test",
                  "stargazers_count": 100,
                },
                Object {
                  "contributors_url": "www.test2.com",
                  "description": "test2 description",
                  "id": 2,
                  "name": "test2",
                  "stargazers_count": 99,
                },
              ]
        }
    />
</div>,
  "nodes": Array [
    <div>
      <Header />
      <RepoList
            repos={
                  Array [
                        Object {
                          "contributors_url": "www.test.com",
                          "description": "test description",
                          "id": 1,
                          "name": "test",
                          "stargazers_count": 100,
                        },
                        Object {
                          "contributors_url": "www.test2.com",
                          "description": "test2 description",
                          "id": 2,
                          "name": "test2",
                          "stargazers_count": 99,
                        },
                      ]
            }
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 61,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "loaded": true,
          "repos": Array [
            Object {
              "contributors_url": "www.test.com",
              "description": "test description",
              "id": 1,
              "name": "test",
              "stargazers_count": 100,
            },
            Object {
              "contributors_url": "www.test2.com",
              "description": "test2 description",
              "id": 2,
              "name": "test2",
              "stargazers_count": 99,
            },
          ],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Header />
          <RepoList
                    repos={
                              Array [
                                        Object {
                                          "contributors_url": "www.test.com",
                                          "description": "test description",
                                          "id": 1,
                                          "name": "test",
                                          "stargazers_count": 100,
                                        },
                                        Object {
                                          "contributors_url": "www.test2.com",
                                          "description": "test2 description",
                                          "id": 2,
                                          "name": "test2",
                                          "stargazers_count": 99,
                                        },
                                      ]
                    }
          />
</div>,
        "_debugID": 62,
        "_renderedOutput": <div>
          <Header />
          <RepoList
                    repos={
                              Array [
                                        Object {
                                          "contributors_url": "www.test.com",
                                          "description": "test description",
                                          "id": 1,
                                          "name": "test",
                                          "stargazers_count": 100,
                                        },
                                        Object {
                                          "contributors_url": "www.test2.com",
                                          "description": "test2 description",
                                          "id": 2,
                                          "name": "test2",
                                          "stargazers_count": 99,
                                        },
                                      ]
                    }
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
